Subject: [PATCH] Redundant waiting time of keyguard unlocking after reboot

LOCKOUT_ATTEMPT_DEADLINE and LOCKOUT_ATTEMPT_TIMEOUT_MS
should be cleared for every successful screen unlock.

diff --git a/packages/Keyguard/src/com/android/keyguard/KeyguardAbsKeyInputView.java b/packages/Keyguard/src/com/android/keyguard/KeyguardAbsKeyInputView.java
index 038e08d..532521c 100644
--- a/packages/Keyguard/src/com/android/keyguard/KeyguardAbsKeyInputView.java
+++ b/packages/Keyguard/src/com/android/keyguard/KeyguardAbsKeyInputView.java
@@ -167,6 +167,7 @@ private void onPasswordChecked(int userId, boolean matched, int timeoutMs,
             mLockPatternUtils.sanitizePassword();
             mCallback.reportUnlockAttempt(userId, true, 0);
             if (dismissKeyguard) {
+                mLockPatternUtils.getLockoutAttemptDeadline(userId);
                 mDismissing = true;
                 mCallback.dismiss(true);
             }
diff --git a/packages/Keyguard/src/com/android/keyguard/KeyguardPatternView.java b/packages/Keyguard/src/com/android/keyguard/KeyguardPatternView.java
index 506f77d..a35daf6 100644
--- a/packages/Keyguard/src/com/android/keyguard/KeyguardPatternView.java
+++ b/packages/Keyguard/src/com/android/keyguard/KeyguardPatternView.java
@@ -288,6 +288,7 @@ private void onPatternChecked(int userId, boolean matched, int timeoutMs,
                 mLockPatternUtils.sanitizePassword();
                 mCallback.reportUnlockAttempt(userId, true, 0);
                 if (dismissKeyguard) {
+                    mLockPatternUtils.getLockoutAttemptDeadline(userId);
                     mLockPatternView.setDisplayMode(LockPatternView.DisplayMode.Correct);
                     mCallback.dismiss(true);
                 }
